cluster-autoscaler/core/scaledown/actuation/actuator.go

```diff
@@ -170,6 +175,33 @@ func (a *Actuator) taintNodesSync(NodeGroupViews []*budgets.NodeGroupView) error
 				return errors.NewAutoscalerError(errors.ApiCallError, "couldn't taint node %q with ToBeDeleted", node)
 			}
 			taintedNodes = append(taintedNodes, node)
+
+			if ind == 0 && time.Now().After(a.ctx.NodeDeleteDelayAfterTaintUpdateTimestamp.Add(nodeDeleteDelayAfterTaintUpdateDuration)) {
+				// Get the newest version of the node.
+				newestNode, err := a.ctx.ClientSet.CoreV1().Nodes().Get(ctx.TODO(), node.Name, metav1.GetOptions{})
+				if err != nil || newestNode == nil {
+					klog.Errorf("Error getting newest version of the node: %v", err)
+					continue
+				}
+				for !taints.HasToBeDeletedTaint(newestNode) {
+					time.Sleep(20 * time.Millisecond)
```

Comment:
Polling like this is going to generate too much traffic to the apiserver. CA should wait for an update on watch instead - listers are already getting this information.

Reply:
Done